core/warning/warn_spec.rb

```diff
@@ -0,0 +1,11 @@
+require File.expand_path("../../../spec_helper", __FILE__)
+
+describe "Warning.warn" do
+  ruby_version_is "2.4" do
+    it "complains" do
+      -> {
+        Warning.warn("Chunky bacon!")
+      }.should complain("Chunky bacon!")
+    end
+  end
+end
```

Comment:
In 2.5, Kernel#warn calls Warning#warn, but on 2.4 it's only called by internal warnings.
So one interesting spec would be to eval some code causing a warning and make sure Warning.warn is called by it with the message.

Reply:
Maybe something like `Warning.should_receive(:warn)` would work.

Reply:
Do nested `ruby_version_is` blocks work?  I currently don't have trunk installed on my laptop

Reply:
Turns out yes!  I'll add two specs for 2.5, one which calls `Kernel.warn` and one which emits a parser warning

Reply:
Awesome!